torch/distributed/_functional_collectives_impl.py

```diff
@@ -92,6 +95,9 @@ def _wait_reg_dec(ptr, wait_reg):
 
 
 def _register_tensor_wrapper(tensor) -> None:
+    if USE_NATIVE_C10D_FUNCTIONAL:
+        # Tensor storage -> work mapping is maintained in C++
+        return
```

Comment:
Question: how we are maintaining the ACT wrapper subclass instance mapping to the work? 

iiuc the current behavior we have:
```
act_tensor = funcol.allreduce(..) # the returned act_tensor.elem map to work, along with `act_tensor` itself
new_act = act_tensor.view(..) # the returned new_act also tracks map to work
```
Our new behavior is that we don't track the tensor wrapper subclass mapping to the work, and we only track the underlying `act.elem` mapping to the work?

Reply:
> Our new behavior is that we don't track the tensor wrapper subclass mapping to the work, and we only track the underlying act.elem mapping to the work?

Hey @wanchaol, in the new impl, we use `tensor.storage().getWeakStorageImpl()` to identify the associated work. The scenario you describe should work. This is a very useful point. I will add a test case for this scenario!